Script worker

erthalion opened the pull request:
Script worker allows much more flexible definition of load structure. Few examples:

```
# short version, only one scenario run
$ strace -f -o /tmp/log berserker -f workloads/example.short.ber
[2025-07-22T13:56:42Z INFO  berserker] Config: WorkloadConfig { restart_interval: 10, per_core: true, workers: 1, workload: Processes { arrival_rate: 10.0, departure_rate: 200.0, random_process: true }, duration: 0 }
; ModuleID = 'main'
source_filename = "main"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128"

@name = private unnamed_addr constant [5 x i8] c"stub\00", align 1
@const = private unnamed_addr constant [14 x i8] c"run task stub\00", align 1
@const.1 = private unnamed_addr constant [20 x i8] c"open file /tmp/test\00", align 1
@const.2 = private unnamed_addr constant [10 x i8] c"/tmp/test\00", align 1

declare i64 @task(i64 %0)

declare i64 @debug(i64 %0)

declare i64 @open(i64 %0)

define i64 @main() {
entry:
  %task = call i64 @debug(ptr @const)
  %task1 = call i64 @task(ptr @name)
  %task2 = call i64 @debug(ptr @const.1)
  %task3 = call i64 @open(ptr @const.2)
  ret i64 0
}
[2025-07-22T13:56:42Z INFO  berserker] Child 201315
[2025-07-22T13:56:42Z INFO  berserker] waitpid: 201315
[2025-07-22T13:56:42Z INFO  berserker] 201315 stopped

$ cat /tmp/log | grep -E test\|stub
202305 execve("/path/to/stub", ["stub", "ZYNnTXs"], 0x7ffd68f58ed8 /* 60 vars */ <unfinished ...>
202303 openat(AT_FDCWD, "/tmp/test", O_WRONLY|O_CREAT|O_CLOEXEC, 0666) = 3
```

```
# long version, running the scenario with specified distribution
$ berserker -f workloads/example.ber
[2025-07-22T14:00:36Z DEBUG berserker] ARGS Args { flag_c: None, flag_f: Some("workloads/example.ber") }
[...]
[2025-07-22T14:00:36Z INFO  berserker] Child 203681
[2025-07-22T14:00:36Z INFO  berserker] waitpid: 203678
[2025-07-22T14:00:36Z DEBUG berserker::worker::script] run task stub
[2025-07-22T14:00:36Z INFO  berserker::worker::script] Interval 0.0586862540339888, rounded 59
[2025-07-22T14:00:36Z DEBUG berserker::worker::script] open file /tmp/test
[2025-07-22T14:00:36Z INFO  berserker::worker::script] Interval 0.10129037145396365, rounded 101
[2025-07-22T14:00:36Z DEBUG berserker::worker::script] run task stub
[2025-07-22T14:00:36Z DEBUG berserker::worker::script] open file /tmp/test
[2025-07-22T14:00:36Z INFO  berserker::worker::script] Interval 0.04452548013581766, rounded 45
[2025-07-22T14:00:36Z DEBUG berserker::worker::script] run task stub
[2025-07-22T14:00:36Z DEBUG berserker::worker::script] open file /tmp/test
[2025-07-22T14:00:36Z INFO  berserker::worker::script] Interval 0.08633732479159864, rounded 86
[...]
```

```
# running with config as before
$ berserker -c workloads/syscalls.toml
[2025-07-22T14:02:43Z DEBUG berserker] ARGS Args { flag_c: Some("workloads/syscalls.toml"), flag_f: None }
[2025-07-22T14:02:43Z INFO  berserker] Config: WorkloadConfig { restart_interval: 10, per_core: true, workers: 1, workload: Syscalls { arrival_rate: 10.0, tight_loop: false, syscall_nr: 39 }, duration: 0 }
[2025-07-22T14:02:43Z INFO  berserker] Child 204588
[2025-07-22T14:02:43Z INFO  berserker] Child 204589
[2025-07-22T14:02:43Z INFO  berserker::worker::syscalls] Process 0 from 0
[2025-07-22T14:02:43Z INFO  berserker::worker::syscalls] Running syscall getpid
[...]
```

erthalion committed:
Script worker

Molter73 reviewed (commented):
Sorry I left so many rust formatted blocks that should've been suggestions instead, the GH UI was making the diff show as only with the last line I selected, so I thought having actual formatting for the comments would be more readable.

Overall looks good, thanks for taking the time implementing this!

erthalion committed:
Add machine section for global state.
Add networking instructions.
Add functions with different set of arguments.

erthalion committed:
Misc improvements

Add helpers to generate random values for workload.
Use BTF to attach fentry/fexit to BPF programs (WIP).
Add SIGTERM hook to finish workers processes.

erthalion committed:
Parser tests

erthalion commented:
Most of it was applied piece by piece in separate PRs
